Sources/PayPalNativeCheckout/PayPalClient.swift

```diff
@@ -33,8 +33,11 @@ public class PayPalClient {
     /// Present PayPal Paysheet and start a PayPal transaction
     /// - Parameters:
     ///   - presentingViewController: the ViewController to present PayPalPaysheet on, if not provided, the Paysheet will be presented on your top-most ViewController
-    ///   - delegate: Completion block to handle buyer's approval, cancellation, error, create order callback, shipping change callback
-    public func start(presentingViewController: UIViewController? = nil, orderID: String, delegate: PayPalDelegate?) async {
+    ///   - createOrder: action to perform when an order has been created
+    public func start(
+        presentingViewController: UIViewController? = nil,
+        createOrder: @escaping PayPalCheckout.CheckoutConfig.CreateOrderCallback
```

Comment:
I'm a little confused about how an `orderID` is used with for our merchants integrating with NXO through Northstar. For `CardClient` and `PayPalWebCheckoutClient` flows, a merchant must provide an `orderID` on their `CardRequest` or `PayPalWebCheckoutRequest`. 

Nor `PayPalClient` (NXO), I'm a little confused why this doesn't follow the same pattern. Also, what is this `createOrder` param for?

Reply:
We need to expose the same functionalites NXO has. NXO allows to start the checkout flow with: an `orderId`, `baToken`, an order object and an `approvalSessionId` (vault).
Thats where the `createOrder` callback comes in place. Inside the callback the merchant specifies with what is going to start the Checkout flow.

Check in `PayPalViewModel` for the methods `checkoutWithOrder`, `checkoutWithOrderId`, `checkoutWithBillingAgreement`, `checkoutBAWithoutPurchase`, `checkoutWithVault`. They are very similiar but when they call `paypalClient.start`, what inside the `createOrder` callback is diferent for each one

Reply:
in noxo, instead of having param for `orderId` it has `PayPalCheckout.CheckoutConfig.CreateOrderCallback` callback to set `orderId` or any other param like `vaultSessionId`, `billingAgreementToken` etc

Reply:
> We need to expose the same functionalites NXO has

Do we have to support all NXO use cases/scenarios? Another question - does our demo need to showcase all of them?

Reply:
@scannillo  this also needs further discussion, from what I see, nxo has lot of functionality and they have their demo app, if needed for demo, we need to create Jiras for functionality in demo